sources/vega/commands/v1/transaction.proto

```diff
@@ -8,13 +8,23 @@ import "vega/commands/v1/validator_commands.proto";
 import "vega/commands/v1/oracles.proto";
 
 message InputData {
-  // A random number used to provided uniqueness and prevents against replay
-  // attack.
+  // A number to provide uniqueness, as to prevent accidental replays and,
+  // in combination with `block_height`, deliberate attacks.
+  // A hash over the transaction uniquely identifies the transaction, thus
+  // the nonce provides uniqueness for otherwise identical transactions. 
+  // Granted all other fields are equal, the nonce can either be a counter
+  // or generated at random to submit multiple transactions within the same
+  // block (see below), without being identified as replays.
+  // Please note that Protocol Buffers do not have a cononical, unique encoding
```

Comment:
Ah, see one typo here "cononical" => "canonical". Will wait for further review

Reply:
```suggestion
  // Please note that Protocol Buffers do not have a canonical, unique encoding
```